libs/librepcb/core/project/board/drc/boarddesignrulecheck.cpp

```diff
@@ -2284,6 +2301,78 @@ QVector<Path> BoardDesignRuleCheck::getViaLocation(
   return {Path::circle(via.size).translated(via.position)};
 }
 
+bool BoardDesignRuleCheck::isViaUseless(const Data& data,
+                                        const Data::Segment& ns,
+                                        const Data::Via& via) noexcept {
+  // The layers of the traces directly connected to the via have already been
+  // added, so if there are two or more connected layers, the via is not
+  // useless and we can skip all the other checks.
+  if (via.connectedLayers.count() >= 2) {
+    return false;
+  }
+
+  QSet<const Layer*> connectedLayers(via.connectedLayers);
+  const ClipperLib::IntPoint viaPosition =
+      ClipperHelpers::convert(via.position);
+
+  // A via is considered to be connected to a plane if it contains
+  // the layer the plane is on, both of them are part of a net, they are in
+  // the same net, and the via is physically within the plane
+  for (const Data::Plane& plane : data.planes) {
+    const int copperNumber = plane.layer->getCopperNumber();
+    if (((via.startLayer->getCopperNumber() > copperNumber) ||
+         (via.endLayer->getCopperNumber() < copperNumber)) ||
+        ((!plane.net) || (!ns.net) || (plane.net != ns.net)) ||
+        connectedLayers.contains(plane.layer)) {
+      continue;
+    }
+
+    const ClipperLib::Path clipperPlanePath =
+        ClipperHelpers::convert(plane.outline, maxArcTolerance());
+    if (ClipperLib::PointInPolygon(viaPosition, clipperPlanePath)) {
+      connectedLayers.insert(plane.layer);
+      if (connectedLayers.count() >= 2) {
+        return false;
+      }
+    }
+  }
+
+  // For each pad geometry that shares a layer with the current via, check if
+  // the center of the via is in the outline of the pad.
+  for (const Data::Device& device : data.devices) {
+    for (const Data::Pad& pad : device.pads) {
+      const Transform transform(pad.position, pad.rotation, pad.mirror);
+
+      for (auto it = pad.geometries.begin(); it != pad.geometries.end(); it++) {
+        const int copperNumber = it.key()->getCopperNumber();
+        if (it.key()->isCopper() ||
```

Comment:
The logic is the wrong way around :see_no_evil: 

```suggestion
        if ((!it.key()->isCopper()) ||
```